membership-attribute-service/app/services/AttributesMaker.scala

```diff
@@ -35,18 +44,36 @@ class AttributesMaker extends LazyLogging {
 
     val hasAttributableProduct = membershipSub.nonEmpty || contributionSub.nonEmpty || subsWhichIncludeDigitalPack.nonEmpty
 
-    hasAttributableProduct.option {
-      val tier: Option[String] = membershipSub.flatMap(getCurrentPlans(_).headOption.map(_.charges.benefits.head.id))
-      val recurringContributionPaymentPlan: Option[String] = contributionSub.flatMap(getTopPlanName)
-      val membershipJoinDate: Option[LocalDate] = membershipSub.map(_.startDate)
-      val latestDigitalPackExpiryDate: Option[LocalDate] = Some(subsWhichIncludeDigitalPack.map(_.termEndDate)).filter(_.nonEmpty).map(_.max)
-      ZuoraAttributes(
-        UserId = identityId,
-        Tier = tier,
-        RecurringContributionPaymentPlan = recurringContributionPaymentPlan,
-        MembershipJoinDate = membershipJoinDate,
-        DigitalSubscriptionExpiryDate = latestDigitalPackExpiryDate
-      )
+    val maybeMembershipAndAccount = membershipSub flatMap { sub: Subscription[AnyPlan] => subsWithAccounts.find(subWithAccount => subWithAccount.subscription == Some(sub)) }
+
+    val maybeAlertAvailable = maybeMembershipAndAccount flatMap { subAndAccount: AccountWithSubscription => subAndAccount.subscription map { sub: Subscription[AnyPlan] =>
+        alertAvailableFor(subAndAccount.account, sub, paymentMethodGetter) map { alertAvailable: Boolean =>
+          if(alertAvailable) Some("membership")
+          else None
+        }
+      }
+    }
+
+    maybeAlertAvailable.getOrElse(Future.successful(None)) map { maybeAction =>
+      maybeAction map { action => SafeLogger.info(s"User $identityId has an action available: $action") }
```

Comment:
should all mentions of 'action' here be 'alert' now?

Reply:
Yes. Thanks for catching that!